### PR TITLE
feat(LERN-95): added getAllStudents for admin endpoint and tests

### PR DESCRIPTION
**Pull request by A1eeee**

### CODE REVIEW

File: backend/src/main/kotlin/ch/learntrack/backend/admin/student/AdminStudentResource.kt [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+@RestController
+@RequestMapping("$ADMIN_ROOT_URL/student")
+public class AdminStudentResource(private val adminStudentService: AdminStudentService) {
+    @PreAuthorize(
+        "@UserAccessAuthorizer.hasUserAccessToSchool(#root.authentication.principal, #schoolId)",
+    )
+    @GetMapping
+    public fun getAllStudentsWithDetailsBySchoolId(@RequestParam schoolId: UUID): List<StudentDetailsDto> =
+        adminStudentService.getAllStudentsForSchool(schoolId)
```
Comment (deepspace1000) [inline backend/src/main/kotlin/ch/learntrack/backend/admin/student/AdminStudentResource.kt:19]:
```suggestion
        adminStudentService.getAllStudentsWithDetailsForSchool(schoolId)
```

File: backend/src/test/kotlin/ch/learntrack/backend/admin/student/StudentIntegrationTest.kt [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+            .expectBodyList(StudentDetailsDto::class.java)
+            .returnResult()
+            .responseBody
+
+        assertThat(response).isNotNull
+        assertThat(response).hasSize(2)
+    }
+
+    @Test
+    fun `should get same grade for different users for assigned user`() {
```
Comment (deepspace1000) [inline backend/src/test/kotlin/ch/learntrack/backend/admin/student/StudentIntegrationTest.kt:91]: Can't see the difference between this test and the one above `should get StudentDetailsDto for assigned school`

Comment (A1eeee) [reply backend/src/test/kotlin/ch/learntrack/backend/admin/student/StudentIntegrationTest.kt:91]: Yeah, idk why I did that😂

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 1  | ❗       |
    


*❗ custom/admin-landing-page*: 


| Before | After |
| --- | --- |
|  |  |

**Review by deepspace1000 — APPROVED** [review]

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 1  | ❗       |
    


*❗ custom/admin-landing-page*: 


| Before | After |
| --- | --- |
|  |  |

**Pull request merged**